components/renku_data_services/project/db.py

```diff
@@ -925,6 +925,20 @@ def __init__(
         self.event_repo: EventRepository = event_repo
         self.session_repo = session_repo
 
+    async def get_project_migrations(
+        self,
+        user: base_models.APIUser,
+    ) -> list[models.ProjectMigrationInfo]:
+        """Get all project migrations from the database."""
+        project_ids = await self.authz.resources_with_permission(user, user.id, ResourceType.project, Scope.READ)
+
+        async with self.session_maker() as session:
+            stmt = select(schemas.ProjectMigrationsORM)
+            stmt = stmt.where(schemas.ProjectMigrationsORM.project_id.in_(project_ids))
+            result = await session.scalars(stmt)
+            project_migrations_orm = result.all()
+            return [p.dump() for p in project_migrations_orm]
```

Comment:
suggestion: please use `stream_scalars` and return an AsyncGenerator from this function. It is a very simple change to make and it is better for the performance of the server.

See `_get_user_namespaces` for example in the namespace component in the db.py file.

Reply:
ok, done in the last commit